Sources/Basics/EnvironmentVariables.swift

```diff
@@ -0,0 +1,50 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import TSCBasic
+import Foundation
+
+public typealias EnvironmentVariables = [String: String]
+
+extension EnvironmentVariables {
+
+    public static func empty() -> EnvironmentVariables {
+        return [:]
+    }
+
+    public static func process() -> EnvironmentVariables {
+        return ProcessInfo.processInfo.environment
+    }
+
+    public mutating func prependPath(_ key: String, value: String) {
+        var values = value.isEmpty ? [] : [value]
+        if let existing = self[key], !existing.isEmpty {
```

Comment:
I've always wondered if this is a performance win in Swift over just `values.append(self[key] ?? [])`.

Reply:
yea the extra `existing.isEmpty` get in the way. not sure about the perf question tho!

Reply:
open to improvement / cleaner way to express this

Reply:
I see — the `existing` here is not an array of strings, but is just a single string.  So if it was already `A:B:C` and you prepend `D`, you have values ending up as `["D", "A:B:C"]`.  I guess that works as long as the values are strictly being joined, and not quoted for example.  I had read this as ending up with `["D", "A", "B", "C"]` which were then rejoined with `:`.  I thought you had done the `isEmpty` as just a performance optimization to avoid an `append` of an array, which wouldn't be needed.  But this is more like an array that's always two elements, like a CAR/CDR Lisp thing if that makes sense.

Given that, I think the code is fine.  It's a bit odd to have an array that's always two elements but I don't think it's worth the expense of splitting the string by `:` only to combine again.

Reply:
> It's a bit odd to have an array that's always two elements but I don't think it's worth the expense of splitting the string by : only to combine again.

right that was my thinking